ea-2021-04-ndvi-automation.py

```diff
@@ -0,0 +1,683 @@
+                            crop_bound=crop_bound_all,
+                            pixel_qa=cl_mask,
+                            vals=vals)
+    
+    mean_ndvi = np.mean(ndvi_clean).values
+    harv_date = pd.to_datetime(date, format='%Y%m%d')
+
+    
+    ndvi_means.append(mean_ndvi)
+    harv_dates.append(harv_date)
+    site_names.append(site_name_each)
+    
+    
+
+all_ndvi_dict = {'site': site_names, 'mean_ndvi': ndvi_means, 'date': harv_dates}
+all_ndvi_mean_df = pd.DataFrame.from_dict(all_ndvi_dict)
+
+all_ndvi_mean_df_date_index = all_ndvi_mean_df.set_index('date')
+
+
+#convert mean_ndvi to floats
+all_ndvi_mean_df_date_index["mean_ndvi"] = all_ndvi_mean_df_date_index["mean_ndvi"].astype(float)
+
+all_ndvi_mean_df_date_index.sort_index(inplace=True)
+
+all_ndvi_mean_df_date_index
+
+
+# In[10]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# In[11]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+# YOUR CODE HERE
+f, ax = plt.subplots(figsize=(10,10))
+
+
+for site, data in all_ndvi_mean_df_date_index.groupby('site'):
+    ax.scatter(data.index, data.mean_ndvi, label=site)
+    
+ax.set(xlabel='Date')
```

Comment:
For time-series with the date series on the x axis, line or bar charts tend to read better than scatter plots

Reply:
I found it very strange looking when I originally plotted as a line graph.